dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/WebContainer.java

```diff
@@ -1568,41 +1580,56 @@ protected void unsetCacheServletWrapperFactory(CacheServletWrapperFactory factor
         // no-op intended here to avoid cacheServletWrapperFactory being null when switching service implementations
     }
 
-    
-    @Reference(service=ServletVersion.class, cardinality=ReferenceCardinality.MANDATORY, policy=ReferencePolicy.DYNAMIC, policyOption=ReferencePolicyOption.GREEDY)
-    protected synchronized void setVersion(ServiceReference<ServletVersion> reference) {
+   /*
+    * Should only be called by loadServletSpecVersion
+    */
```

Comment:
I ran into a problem with the 3.1 Factory unit tests. They failed since no version was set, and I fixed this by adding: 
`webContainer.setVersion(version);` to ServletConfigMock.java See Here: 
https://github.com/OpenLiberty/open-liberty/pull/20004/files/5aac770b82f05f9bf70d2c120f8ed367f9a149db#diff-14401b3ea97e233e3367fe5d66e798161a764cc2beda10da995da558a416cfcfR91

If I change setVersion to public, the tests will fail. Should I remove them instead?

```
    WebFragmentMergerTest. testInjectionTargetEjblocal30
    WebFragmentMergerTest. testInjectionTargetEjbref30
    WebFragmentMergerTest. testInjectionTargetEnvEntry30
    WebFragmentMergerTest. testInjectionTargetMsgRef30
    WebFragmentMergerTest. testInjectionTargetPersConRef30
    WebFragmentMergerTest. testInjectionTargetPersUnitRef30
    WebFragmentMergerTest. testInjectionTargetResEnvRef30
    WebFragmentMergerTest. testInjectionTargetResRef30
    WebFragmentMergerTest. testInjectionTargetServiceRef30

```



Alternatively, should I keep this method as public?

Reply:
Methods updated to protected & i fixed unit test by using reflection.